reporters/reporter-default/src/main/java/org/radargun/reporting/html/HtmlReporter.java

```diff
@@ -147,47 +172,109 @@ private void writeTestReportDocuments(Set<String> combinedTests, Map<String, Lis
+            root.put("testReport", testReport);
+            root.put("enums", DefaultObjectWrapper.getDefaultInstance().getEnumModels());
+            reportTemplate.process(root, printWriter);
+            printWriter.close();
+         } catch (Exception e) {
+            log.error("Templating exception", e);
          }
       }
    }
 
+
    private void writeTimelineDocuments(Collection<Report> reports) {
       for (Report report : reports) {
          String configName = report.getConfiguration().name;
-
          TimelineDocument timelineDocument = new TimelineDocument(timelineConfig, targetDir,
-               configName + "_" + report.getCluster().getClusterIndex(), configName + " on " + report.getCluster(), report.getTimelines(), report.getCluster());
+                 configName + "_" + report.getCluster().getClusterIndex(), configName + " on " + report.getCluster(), report.getTimelines(), report.getCluster());
+
+         freemarker.template.Configuration cfg = initConfig();
          try {
-            timelineDocument.open();
-            timelineDocument.writeTimelines();
-         } catch (IOException e) {
-            log.error("Failed to create timeline report " + configName, e);
-         } finally {
-            timelineDocument.close();
+            timelineDocument.createReportDirectory();
+            timelineDocument.generateTimelineCharts();
+
+            Template reportTemplate = cfg.getTemplate("timelineReport.ftl");
+            PrintWriter printWriter = new PrintWriter(targetDir + File.separator + timelineDocument.getFileName());
+
+            Map root = new HashMap();
+            root.put("timelineDocument", timelineDocument);
+            reportTemplate.process(root, printWriter);
+            printWriter.close();
+         } catch (Exception e) {
+            log.error("Templating exception", e);
          }
       }
    }
 
    private void writeIndexDocument(Collection<Report> reports, Set<String> allTests) {
       IndexDocument index = new IndexDocument(targetDir);
+      index.createReportDirectory();
+      index.prepareDocument(reports);
+      freemarker.template.Configuration cfg = initConfig();
       try {
-         index.open();
-         index.writeTests(allTests);
-         index.writeConfigurations(reports);
-         index.writeScenario(reports);
-         index.writeTimelines(reports);
-         index.writeFooter();
-      } catch (IOException e) {
-         log.error("Failed to write HTML report.", e);
-      } finally {
-         index.close();
+         Template indexTemplate = cfg.getTemplate("index.ftl");
+         PrintWriter printWriter = new PrintWriter(targetDir + File.separator + "index.html");
+
+         Map root = new HashMap();
+         root.put("reporter", this);
+         root.put("indexDocument", index);
+
+         copyResources("style.css");
+         copyResources("script.js");
+         indexTemplate.process(root, printWriter);
+         printWriter.close();
```

Comment:
This one can be included in https://github.com/radargun/radargun/pull/248/files#r35864832 as well.